tests/components/wyoming/test_satellite.py

```diff
@@ -613,6 +613,7 @@ async def test_satellite_error_during_pipeline(hass: HomeAssistant) -> None:
         "homeassistant.components.wyoming.satellite.assist_pipeline.async_pipeline_from_audio_stream",
     ) as mock_run_pipeline:
         await setup_config_entry(hass)
+        await hass.async_block_till_done()
```

Comment:
There's already a block till done inside `setup_config_entry`. There is likely scheduling happening during integration setup that isn't awaited in a controlled fashion in the tests, requiring multiple yields to the event loop here. Not great.

Reply:
Spot on. Likely sleep(0) will also fix it, but its brittle either way

Reply:
Looks like it relies on 
```
        # Start satellite communication
        entry.async_create_background_task(
            hass,
            item.satellite.run(),
            f"Satellite {satellite_info.name}",
        )

```

Reply:
I adjusted it so it waits for the `AsyncTcpClient` creation call to happen before waiting on the other events.